src/metabase/models/serialization.clj

```diff
@@ -1631,15 +1631,24 @@
 
 ;;; ## Utilities
 
+(defn strip-error
+  "Transforms the error in a list of strings to log"
+  [e prefix]
+  (vec
+   (for [[e prefix] (map vector
+                         (take-while some? (iterate #(.getCause ^Exception %) e))
+                         (cons prefix (repeat "  caused by")))]
+     (str (when prefix (str prefix ": "))
+          (ex-message e)
+          (when-let [data (-> (ex-data e)
+                              (dissoc :toucan2/context-trace)
+                              not-empty)]
+            (str " " (pr-str data)))))))
+
 (defmacro log-stripped-error
   "Log errors with no stacktrace"
-  [prefix e]
-  `(loop [prefix# ~prefix
-          e#      ~e]
-     (when e#
-       (log/errorf (str prefix# ": " (ex-message e#) " " (-> (ex-data e#)
-                                                             (dissoc :toucan2/context-trace))))
-       (recur "  caused by" (.getCause ^Exception e#)))))
+  [e prefix]
+  `(run! #(log/error %) (strip-error ~e ~prefix)))
```

Comment:
Reason not to rather log multi line message? I guess some grepability, but it feels dirty.

Reply:
I'm afraid it's going to look bad/will be hard to find in Grafana. Not sure how this will work.

Reply:
Maybe I just need to find some exception and if it's ok this macro can go away.